pkg/pillar/docs/failover.md

```diff
@@ -0,0 +1,181 @@
+# Application failover and volume data protection support
+
+## Overview
+
+Edge devices can be clustered together if they are installed with version of eve which supports kubevirt virtualization.
+The volumes created on those devices are synchronously replicated within the cluster for data protection and high availability. The applications deployed on those devices are automatically failed over to surviving nodes by the underlying kubernetes infrastructure. This document covers the process of failover and the underlying data structures.
+
+## Components
+
+### Block Volumes
+
+In a clustered setup block volumes are treated as a cluster wide objects. Controller picks one of the nodes in the cluster as the designated node id for that volume. Controller sends volume config to all the devices in the cluster with designated node id set to uuid of the device which is supposed to be designated node id for that volume.
+
+EVE API has been enhanced to include Designated node id as String.
+Please look at eve-api/proto/config/storage.proto
+message Volume {
```

Comment:
All the code definitions in this file needs some markdown to make it be rendered correctly.

Reply:
ok

Reply:
Done and will be in updated commit

Reply:
Still doesn't render correctly.